src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -664,7 +670,23 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return undefined;
 	}
 
-	public insertCell(cell: ICellModel, index?: number): ICellModel | undefined {
+	public mergeCells(cell: ICellModel, secondCell: ICellModel): void {
+		let index = this._cells.findIndex(cell => cell.equals(cell));
+		if (index > -1) {
+			cell.source = cell.source.concat(...secondCell.source);
+			this._contentChangedEmitter.fire({
```

Comment:
Do we not need to fire an event for deleting the other cell as well?

Reply:
The deleteCell method fires the event